cpp/src/Ice/Instance.cpp

```diff
@@ -103,6 +104,36 @@ namespace
         }
     }
 
+    string getProgramName()
+    {
+        string programName{"IceC++"};
+#if defined(__APPLE__) && TARGET_OS_IPHONE == 0
+        vector<char> buffer(PATH_MAX);
+        int len = proc_pidpath(getpid(), buffer.data(), static_cast<uint32_t>(buffer.size()));
+        if (len > 0)
+        {
+            programName = string{buffer.data(), static_cast<size_t>(len)};
+        }
+#elif defined(__linux__)
+        // Read the command line from proc
+        string procExePath{"/proc/self/exe"};
+        vector<char> buffer(PATH_MAX);
+        ssize_t len = readlink(procExePath.c_str(), buffer.data(), buffer.size());
+        if (len > 0)
+        {
+            programName = string{buffer.data(), static_cast<size_t>(len)};
+        }
+#elif defined(_WIN32)
+        vector<char> buffer(MAX_PATH);
+        DWORD len = GetModuleFileNameA(NULL, buffer.data(), static_cast<DWORD>(buffer.size()));
+        if (len > 0 && len < buffer.size())
```

Comment:
From the API reference:
https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamea

I don't see how the returned len can be > MAX_PATH. So what's special about len = MAX_PATH that you want to silently fail?

Reply:
What's the point for this `len < buffer.size()` since it's always true?